src/tests/ftest/util/telemetry_utils.py

```diff
@@ -486,8 +496,12 @@ def list_metrics(self, hosts=None):
         host_list = hosts or self.hosts
         self.log.info("Listing telemetry metrics from %s", host_list)
         for host in host_list:
-            data = self.dmg.telemetry_metrics_list(host=host)
             info[host] = []
+            try:
+                data = self.dmg.telemetry_metrics_list(host=host)
+            except CommandFailure as e:
+                self.log.error("Failed to list metrics on %s: %s", host, e)
+                continue
```

Comment:
```suggestion
            except CommandFailure as err:
                self.log.error("Failed to list metrics on %s: %s", host, err)
                continue
```

Also, do we really want to just skip this failure?

Reply:
We're not just skipping, though. We're logging it and proceeding to attempt to collect metrics from other hosts in the hostlist. If the intended default behavior of this function is that it shouldn't succeed unless all hosts return the expected metrics, then I could add a bool to enable this as optional behavior. IMO that's kind of pointless and less flexible, because any callers should be sanity checking the results, and if they care specifically about whether or not every host in the results has data, then they should check for that.

Reply:
Okay I'm inclined to agree with you. Looks like current usages do verify in some way or would otherwise fail. And obviously if a test was previously getting an exception it should already be failing.